### PR TITLE
Fix revoked client

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: src/fetch.js [hunk 1]
```diff
@@ -99,19 +99,25 @@ function handleJSONResponse (res) {
   }
 }
 
-export class FetchError {
+export class FetchError extends Error {
   constructor (res, reason) {
+    super()
+    Error.captureStackTrace(this, this.constructor)
```
Comment (jinroh) [inline src/fetch.js]: I think `Error.captureStackTrace` is not standard (not available in firefox). We should have a guard `if (Error.captureStackTrace) { .. }`

File: src/auth_v3.js [hunk 2]
```diff
@@ -281,8 +281,8 @@ export function oauthFlow (cozy, storage, clientParams, onRegistered) {
           //
           // If it is an explicit Unauthorized error though, we bail, clear th
           // cache and retry.
-          if (FetchError.isUnauthorized(err)) {
-            throw err
+          if (FetchError.isUnauthorized(err) || FetchError.isNotFound(err)) {
+            throw new Error('Client has been revoked')
```
Comment (enguerran) [inline src/auth_v3.js]: Why don't you throw a `new FetchError`?

Comment (nono) [inline src/auth_v3.js]: I don't consider that a revoked client is a `FetchError`: it is not really tied to a single request to the stack, but something more global.

Comment (enguerran) [inline src/auth_v3.js]: Sounds good to me.

**Pull request merged**